test/framework/src/main/java/org/elasticsearch/transport/nio/channel/ChannelConsumerAdaptor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.transport.nio.channel;
+
+import org.elasticsearch.action.ActionListener;
+
+import java.util.function.Consumer;
+
+public class ChannelConsumerAdaptor implements ActionListener<NioChannel> {
```

Comment:
hmm can't we do:

```Java
NioChannel channel = ...;
Consumer<NioChannel> consumer = ...;
ActionListener<NioChannel> insteadOfAdaptor = ActionListener.wrap(consumer, e -> consumer.accept(channel));
```

instead of having this adaptor class?